### PR TITLE
monorepo: convert test data from JSON to objects

### LINKED ISSUE #3637: Migrate JSON testing data to TypeScript objects
In order to improve type recognition and imports, we want to migrate all JSON testing data to typescript objects. This will notably allow literal string type recognition (e.g. the commonly-used PrefixedHexString), which is not possible when importing JSON. 

See discussions from https://github.com/ethereumjs/ethereumjs-monorepo/pull/3621

Packages converted:

- [x] Block (https://github.com/ethereumjs/ethereumjs-monorepo/pull/3642)
- [x] Blockchain (https://github.com/ethereumjs/ethereumjs-monorepo/pull/3642)
- [x] Client (https://github.com/ethereumjs/ethereumjs-monorepo/pull/3653)
- [x] Common (https://github.com/ethereumjs/ethereumjs-monorepo/pull/3642)
- [x] Devp2p (https://github.com/ethereumjs/ethereumjs-monorepo/pull/3642)
- [x] Ethash (https://github.com/ethereumjs/ethereumjs-monorepo/pull/3680)
- [x] Evm (https://github.com/ethereumjs/ethereumjs-monorepo/pull/3681)
- [x] Genesis (N/A)
- [x] RLP (https://github.com/ethereumjs/ethereumjs-monorepo/pull/3682)
- [x] StateManager (https://github.com/ethereumjs/ethereumjs-monorepo/pull/3660)
- [x] Trie (https://github.com/ethereumjs/ethereumjs-monorepo/pull/3683)
- [x] Tx (https://github.com/ethereumjs/ethereumjs-monorepo/pull/3684)
- [x] Util (https://github.com/ethereumjs/ethereumjs-monorepo/pull/3685)
- [x] Verkle (N/A)
- [x] Vm (https://github.com/ethereumjs/ethereumjs-monorepo/pull/3686)
- [x] Wallet (N/A)

### PR DESCRIPTION
**Pull request by gabrocheleau**
This PR begins addressing #3637 by converting JSON test data to objects.

The packages fully converted by this PR are:
- Block
- Blockchain
- Devp2p

Some other packages have been partially migrated because of cross-dependencies.

### CODE REVIEW

File: packages/block/src/from-beacon-payload.ts [hunk 1]
```diff
@@ -39,17 +39,17 @@ export type BeaconPayloadJSON = {
...
   logs_bloom: PrefixedHexString
   prev_randao: PrefixedHexString
-  block_number: PrefixedHexString
-  gas_limit: PrefixedHexString
-  gas_used: PrefixedHexString
-  timestamp: PrefixedHexString
+  block_number: string
+  gas_limit: string
+  gas_used: string
+  timestamp: string
```
Comment (holgerd77) [inline packages/block/src/from-beacon-payload.ts:45]: What is going on here ?!

**Comment by holgerd77** [comment]
🙏

File: packages/block/src/from-beacon-payload.ts (hunk 1 above)
Comment (gabrocheleau) [reply packages/block/src/from-beacon-payload.ts:45]: Through this work I realized that these values are actually numeric string, not 0xprefixed strings. See for e.g.

https://github.com/ethereumjs/ethereumjs-monorepo/blob/monorepo/typescript-test-files/packages/block/test/testdata/payload-slot-87335.ts

Comment (gabrocheleau) [reply packages/block/src/from-beacon-payload.ts]: Created a numeric string type for it

Comment (holgerd77) [reply packages/block/src/from-beacon-payload.ts]: Oh, that's super cool (so: the new type)! Again on this level, small change, but really another typing milestone in our libraries, once we have applied this consistently throughout the libraries! 🤩

**Review by acolytec3 — APPROVED** [review]
LGTM

**Pull request merged**